legend-engine-language-pure-grammar/src/main/java/org/finos/legend/engine/language/pure/grammar/from/data/embedded/ModelStoreDataParseTreeWalker.java

```diff
@@ -49,11 +53,13 @@
 
     private final ParseTreeWalkerSourceInformation walkerSourceInformation;
     private final SourceInformation sourceInformation;
+    private final PureGrammarParserExtensions extensions;
 
-    public ModelStoreDataParseTreeWalker(ParseTreeWalkerSourceInformation walkerSourceInformation, SourceInformation sourceInformation)
+    public ModelStoreDataParseTreeWalker(ParseTreeWalkerSourceInformation walkerSourceInformation, SourceInformation sourceInformation, PureGrammarParserExtensions extensions)
     {
         this.walkerSourceInformation = walkerSourceInformation;
         this.sourceInformation = sourceInformation;
+        this.extensions = extensions;
```

Comment:
Do we need extensions here, I believe DataElementRef and EmbeddedData are both in core, right?

Reply:
parseEmbeddedData functions requires an extension to parse the data, also doesnt it lie in coreCompilerExtensions

Reply:
Resolved